projects/noloan/admin/app/src/main/java/hasadna/noloan/admin/app/firestore/FirestoreClient.java

```diff
@@ -0,0 +1,118 @@
+package hasadna.noloan.admin.app.firestore;
+
+import android.util.Base64;
+import android.util.Log;
+
+import com.google.android.gms.tasks.TaskCompletionSource;
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.DocumentChange;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.MessageLite;
+
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+
+import hasadna.noloan.admin.app.DBMessagesHolder;
+import hasadna.noloan.protobuf.SmsProto.SmsMessage;
+
+public class FirestoreClient {
+  public static final String SPAM_COLLECTION_PATH = "noloan/spam/sms";
+
+  public static final String USER_SUGGEST_COLLECTION = "noloan/spam/suggestion";
+
+  private FirebaseFirestore client;
+
+  public FirestoreClient() {
+    client = FirebaseFirestore.getInstance();
+  }
+
+  public void writeMessage(SmsMessage message, String path) {
+    FirestoreElement element = encodeMessage(message);
+    client.collection(path).add(element);
+  }
+
+  public void deleteMessage(SmsMessage sms, String path) {
+    client.collection(path).document(sms.getID()).delete().
+        addOnSuccessListener(aVoid -> {
+          Log.d("BLAAAAA","!!!!!!!!!!!!!!!!! success");
```

Comment:
? :-)

Reply:
I said it need clean up

Reply:
What do we want to happen on success and failed of deleted message from the DB?

Reply:
On success - nothing.
On fail - log and show message to user.